pkg/cloudprovider/provider/openstack/helper.go

```diff
@@ -414,18 +414,15 @@ NetworkLoop:
 }
 
 func getDefaultSubnet(client *gophercloud.ProviderClient, network *osnetworks.Network, region string) (*string, error) {
-	if len(network.Subnets) == 0 {
-		return nil, nil
-	} else if len(network.Subnets) == 1 {
+	if len(network.Subnets) == 1 {
```

Comment:
What about: 
```
if len(network.Subnets) > 0 {
  return &network.Subnets[0], nil
} else {
.......
}
```

Reply:
I explicitly did not want to rigorously change the semantics for the case that `network.Subnets` has exactly one element already.
But perhaps we should clarify first if API-requesting sub-networks brings more value (more recent data or a different set of sub-networks than available in the networks structs).